joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala

```diff
@@ -1265,6 +1265,24 @@ class AstCreator(
 
     methodNames.put(methodNode.name, methodFullName)
     val blockNode = NewBlock().typeFullName(Defines.Any)
+
+    /* Before creating ast, we traverse the method params and identifiers and link them*/
+    val identifiers =
+      astBody.flatMap(ast => ast.nodes.filter(_.isInstanceOf[NewIdentifier])).asInstanceOf[Seq[NewIdentifier]]
+
+    astMethodParamSeq
+      .flatMap(ast =>
+        ast.nodes
+          .filter(_.isInstanceOf[NewMethodParameterIn])
+          .asInstanceOf[Seq[NewMethodParameterIn]]
+      )
+      .foreach(paramNode => {
+        val linkIdentifiers = identifiers.filter(_.name.equals(paramNode.name))
```

Comment:
Pass by value works here too
```suggestion
        val linkIdentifiers = identifiers.filter(_.name == paramNode.name)
```